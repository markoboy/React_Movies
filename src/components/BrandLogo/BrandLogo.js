/* eslint-disable react/jsx-one-expression-per-line */
import PropTypes from 'prop-types';
import React, { memo } from 'react';
import './BrandLogo.scss';

function BrandLogo({ name, subName }) {
  return (
    <h1 className="brand-logo">
      <a href="/">
        {name}<span className="font--regular">{subName}</span>
      </a>
    </h1>
  );
}

BrandLogo.defaultProps = {
  name: 'netflix',
  subName: 'roulette',
};

BrandLogo.propTypes = {
  name: PropTypes.string,
  subName: PropTypes.string,
};

export default memo(BrandLogo);
