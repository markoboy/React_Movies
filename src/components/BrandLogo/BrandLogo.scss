@import 'essential';

.brand-logo {
  color: $color-primary;
  font-weight: $font-weight-bold;
  font-size: 1.5em;
  margin: 0;
}
