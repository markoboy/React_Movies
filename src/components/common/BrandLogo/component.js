/* eslint-disable react/jsx-one-expression-per-line */
import PropTypes from 'prop-types';
import React, { memo } from 'react';
import './styles.scss';

function BrandLogoComponent({ name, subName }) {
  return (
    <h1 className="brand-logo">
      <a href="/">
        {name}<span className="font--regular">{subName}</span>
      </a>
    </h1>
  );
}

BrandLogoComponent.defaultProps = {
  name: 'netflix',
  subName: 'roulette',
};

BrandLogoComponent.propTypes = {
  name: PropTypes.string,
  subName: PropTypes.string,
};

export default memo(BrandLogoComponent);
