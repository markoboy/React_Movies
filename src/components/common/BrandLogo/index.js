export { default } from './BrandLogo.component';
