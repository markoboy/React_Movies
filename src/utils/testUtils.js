/* eslint-disable import/no-extraneous-dependencies */
import defaultStore from '@store/index';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export function renderWithStore(
  ui,
  { store = defaultStore, route = '/', ...renderOptions } = {}
) {
  window.history.pushState({}, 'Test page', route);

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }

  return render(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

export function renderWithRouter(ui, { route = '/' } = {}) {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: BrowserRouter });
}
