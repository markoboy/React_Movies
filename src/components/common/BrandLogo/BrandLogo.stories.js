import React from 'react';

import BrandLogo from './index';

export default {
  component: BrandLogo,
  title: 'Common/BrandLogo',
};

const Template = (args) => <BrandLogo {...args} />;

export const Default = Template.bind({});
Default.args = {
  name: 'not netflix',
  subName: 'roulette',
};
