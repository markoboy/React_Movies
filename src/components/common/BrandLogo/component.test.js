import React from 'react';
import { renderWithRouter } from '@utils/testUtils';

import BrandLogo from './component';

describe('BrandLogo Component', () => {
  it('renders BrandLogo Component with provided text', () => {
    const { container, getByText } = renderWithRouter(<BrandLogo name="netflix" subName="roulette" />);

    const header = getByText(/netflix/);

    expect(header).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
