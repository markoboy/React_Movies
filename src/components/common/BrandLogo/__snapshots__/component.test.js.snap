// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandLogo Component renders BrandLogo Component with provided text 1`] = `
<div>
  <h1
    class="brand-logo"
  >
    <a
      href="/"
    >
      netflix
      <span
        class="font--regular"
      >
        roulette
      </span>
    </a>
  </h1>
</div>
`;
